dds/DdsDcpsTypeSupportExt.idl

```diff
@@ -39,6 +39,11 @@ module OpenDDS {
   #endif
 
       boolean has_dcps_key();
+
+      ::DDS::ReturnCode_t unregister_type(
+          in ::DDS::DomainParticipant domain,
+          in string type_name);
```

Comment:
Interesting, that does seem to be allowed by the spec.  I wonder why it doesn't just use two different objects (of the same class) for that?  Perhaps something that was "lost in translation" when they attempted to model static member functions in IDL.  In any case, we're stuck with it now.

Given that, the implementation of TypeSupportImpl::get_type_name() should probably work slightly differently.  But it would probably only matter in very strange use cases.